src/translations/mapCollection.js

```diff
@@ -0,0 +1,114 @@
+import { isUndefined, isDefined } from '../helpers/mixed';
+import { mixin } from '../helpers/object';
+import localHooks from '../mixins/localHooks';
+
+// Counter for checking if there is a memory leak.
+let registeredMaps = 0;
+
+class MapCollection {
+  constructor() {
+    this.mappings = new Map();
+  }
+
+  /**
+   * Register custom index map.
+   *
+   * @param {String} uniqueName Unique name of the indexes list.
+   * @param {BaseMap} map Map containing miscellaneous (i.e. meta data, indexes sequence), updated after remove and insert data actions.
+   * @returns {BaseMap|undefined}
+   */
+  register(uniqueName, map) {
+    if (this.mappings.has(uniqueName) === false) {
+      this.mappings.set(uniqueName, map);
+
+      map.addLocalHook('change', () => this.runLocalHooks('change', map));
+
+      registeredMaps += 1;
+    }
+  }
+
+  /**
+   * Unregister custom index map.
+   *
+   * @param {String} name Name of the indexes list.
```

Comment:
Inconsistency: this JSDoc uses the words "map" and "list" as synonyms. Should be `Name of the map.`.